File: Exercism/javascript/word-count/word-count.js
'use strict'

export class Words {
  count (sentence) {
    return sentence
      .trim()
      .toLowerCase()
      .split(/\s+/)
      .reduce((obj, word) => {
        obj[word] = obj.hasOwnProperty(word)
          ? obj[word] + 1
          : 1
        return obj
      }, {})
  }
}

File: Exercism/python/word-count/word_count.py
import re


def count_words(sentence: str) -> dict:
    """Counts the occurrences of each word in a phrase."""
    words = re.findall(r"[a-z0-9]+(?:'[a-z]+)?", sentence.lower())
    return {
        wrd: words.count(wrd)
        for wrd in set(words)
    }
